### PR TITLE
SFM Plus Feature Addition - Heading Hold

### PR DESCRIPTION
**Pull request by BradMick**

### CODE REVIEW

File: @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/XEH_preInit.sqf [hunk 1]
```diff
@@ -117,7 +117,7 @@ fza_ah64_salvofired = 0;
 fza_ah64_mynum = 0;
 fza_ah64_lastdir = 0;
 fza_ah64_dps = 0;
-fza_ah64_slip = 0;
+fza_ah64_sideslip = 0;
```
Comment (mattysmith22) [inline @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/XEH_preInit.sqf:120]: Could the renaming of `slip` to `sideslip` go separately please

File: @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/core/fn_coreGetWCAs.sqf [hunk 2]
```diff
@@ -263,5 +263,8 @@ if (_heli getVariable "fza_ah64_rtrbrake") then {
 if (_fcrDamage >= SYS_FCR_DMG_THRESH) then {
     _wcas pushBack [WCA_ADVISORY, "FCR FAULT", "FCR FAULT"];
 };
+if (isTouchingGround _heli) then {
```
Comment (mattysmith22) [inline @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/core/fn_coreGetWCAs.sqf:266]: I still don't know how I didn't realise this command existed ... good job

File: @AH-64D Apache Longbow/Addons/fza_ah64_sfmplus/functions/fmc/fn_fmcAttitudeHold.sqf [hunk 3]
```diff
@@ -20,7 +20,7 @@ if ( _heli getVariable "fza_ah64_attHoldActive" && !(_heli getVariable "fza_ah64
     if (_subMode == "pos") then {
 
         private _roll  = [_pidRoll,  _deltaTime, 0.0, _curVelX] call fza_fnc_pidRun;    //(_curVelX + _distX) / 2.0
-        _roll          = [_roll, -1.0, 1.0] call BIS_fnc_clamp;
+        _roll          = [_roll,  -1.0, 1.0] call BIS_fnc_clamp;
```
Comment (mattysmith22) [inline @AH-64D Apache Longbow/Addons/fza_ah64_sfmplus/functions/fmc/fn_fmcAttitudeHold.sqf:23]: Why was this changed?

File: @AH-64D Apache Longbow/Addons/fza_ah64_sfmplus/functions/fmc/fn_fmcHeadingHold.sqf [hunk 4]
```diff
@@ -0,0 +1,72 @@
+params ["_heli", "_deltaTime"];
+#include "\fza_ah64_sfmplus\headers\core.hpp"
+
+private _pidHdg     = _heli getVariable "fza_sfmplus_pid_hdgHold";
+private _pidTrn     = _heli getVariable "fza_sfmplus_pid_trnCoord";
+//_pidTrn set ["kp", KP];
+//_pidTrn set ["ki", KI];
+//_pidTrn set ["kd", KD];
```
Comment (mattysmith22) [inline @AH-64D Apache Longbow/Addons/fza_ah64_sfmplus/functions/fmc/fn_fmcHeadingHold.sqf:8]: Please can you remove these comments?

File: @AH-64D Apache Longbow/Addons/fza_ah64_sfmplus/functions/fmc/fn_fmcHeadingHold.sqf [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+private _curHdg     = getDir _heli;
+private _desiredHdg = _heli getVariable "fza_ah64_hdgHoldDesiredHdg";
+private _hdgError   = [_curHdg - _desiredHdg] call CBA_fnc_simplifyAngle180;
+private _curVel     = vectorMagnitude [velocityModelSpace _heli # 0, velocityModelSpace _heli # 1];
+private _subMode    = _heli getVariable "fza_ah64_hdgHoldSubMode";
+private _attSubMode = _heli getVariable "fza_ah64_attHoldSubMode";
+private _output     = 0.0;
+//Breakout values expand as the aircraft goes faster to provide good pedal response
+//at a hover. The expanded range is meant to de-sensitize the pedasl in order to
```
Comment (mattysmith22) [inline @AH-64D Apache Longbow/Addons/fza_ah64_sfmplus/functions/fmc/fn_fmcHeadingHold.sqf:17]: Good documentation, please can you fix the typo in "pedal"?

File: @AH-64D Apache Longbow/Addons/fza_ah64_sfmplus/functions/fmc/fn_fmcHeadingHold.sqf [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+};
+if (_attSubMode == "att" || _curVel > HDG_HOLD_SPEED_SWITCH_ACCEL) then {
+    _breakoutValue = ATT_HOLD_BREAKOUT_VALUE;
+};
+//If we are on the ground, or if the force trim is interupted, or the pilot has exceeded
+//the breakout values for the pedals, then heading hold is not active (doing work)
+//otherwise, heading hold is ALWAYS active
+if (isTouchingGround _heli
+    || _heli getVariable "fza_ah64_forceTrimInterupted" 
+    || fza_sfmplus_pedalLeftRight <= -_breakoutValue
```
Comment (mattysmith22) [inline @AH-64D Apache Longbow/Addons/fza_ah64_sfmplus/functions/fmc/fn_fmcHeadingHold.sqf:34]: Could this be simplified to `abs fza_sfmplus_pedalLeftRight <= _breakoutValue`

**Review by mattysmith22 — CHANGES_REQUESTED** [review]

File: @AH-64D Apache Longbow/Addons/fza_ah64_sfmplus/functions/fmc/fn_fmcHeadingHold.sqf (hunk 6 above)
Comment (BradMick) [reply @AH-64D Apache Longbow/Addons/fza_ah64_sfmplus/functions/fmc/fn_fmcHeadingHold.sqf:34]: Pedals are a -1 to 1 value not 0 to 1 value

File: @AH-64D Apache Longbow/Addons/fza_ah64_sfmplus/functions/fmc/fn_fmcAttitudeHold.sqf (hunk 3 above)
Comment (BradMick) [reply @AH-64D Apache Longbow/Addons/fza_ah64_sfmplus/functions/fmc/fn_fmcAttitudeHold.sqf:23]: To align with the values below

File: @AH-64D Apache Longbow/Addons/fza_ah64_sfmplus/functions/fmc/fn_fmcHeadingHold.sqf (hunk 5 above)
Comment (BradMick) [reply @AH-64D Apache Longbow/Addons/fza_ah64_sfmplus/functions/fmc/fn_fmcHeadingHold.sqf:17]: Fixed

File: @AH-64D Apache Longbow/Addons/fza_ah64_sfmplus/functions/fmc/fn_fmcHeadingHold.sqf (hunk 4 above)
Comment (BradMick) [reply @AH-64D Apache Longbow/Addons/fza_ah64_sfmplus/functions/fmc/fn_fmcHeadingHold.sqf:8]: Removed

File: @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/core/fn_coreGetWCAs.sqf (hunk 2 above)
Comment (BradMick) [reply @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/core/fn_coreGetWCAs.sqf:266]: Right? Randomly happened across it

File: @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/XEH_preInit.sqf (hunk 1 above)
Comment (BradMick) [reply @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/XEH_preInit.sqf:120]: It ensures its the same name across the entire use case

Comment (mattysmith22) [reply @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/XEH_preInit.sqf:120]: I'll leave this one as it is somewhat related

**Review by mattysmith22 — CHANGES_REQUESTED** [review]

File: @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/core/fn_coreGetWCAs.sqf (hunk 2 above)
[the remark this answers is not in the document]
Comment (BradMick) [reply @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/core/fn_coreGetWCAs.sqf:266]: It is related in that it's the indication that alerts the pilot to the fact that heading hold is enabled because the helicopter is no longer on the squat (weight on wheels) switch

**Review by mattysmith22 — APPROVED** [review]
